File: app/src/main/java/com/phlox/tvwebbrowser/activity/main/view/WebTabItemView.java
package com.phlox.tvwebbrowser.activity.main.view;

import android.content.Context;
import android.net.Uri;
import android.text.TextUtils;
import android.view.LayoutInflater;
import android.view.View;
import android.widget.Button;
import android.widget.FrameLayout;
import android.widget.ImageView;
import android.widget.LinearLayout;
import android.widget.TextView;

import com.phlox.tvwebbrowser.R;
import com.phlox.tvwebbrowser.model.WebTabState;

/**
 * Created by PDT on 24.08.2016.
 */
public class WebTabItemView extends FrameLayout {
    private WebTabState tabState;
    private Listener listener;

    private final TextView tvTitle;
    private final ImageView ivThumbnail;
    private final LinearLayout llContainer;
    private final Button btnClose;

    public interface Listener {
        void onTabSelected(WebTabState tab);
        void onTabDeleteClicked(WebTabState tab);
        void onNeededThumbnailSizeCalculated(int width, int height);
    }

    public WebTabItemView(Context context) {
        super(context);
        LayoutInflater.from(context).inflate(R.layout.view_tab_item, this);
        tvTitle = (TextView) findViewById(R.id.tvTitle);
        ivThumbnail = (ImageView) findViewById(R.id.ivThumbnail);
        llContainer = (LinearLayout) findViewById(R.id.llContainer);
        btnClose = (Button) findViewById(R.id.btnClose);

        llContainer.setOnClickListener(new OnClickListener() {
            @Override
            public void onClick(View view) {
                listener.onTabSelected(tabState);
            }
        });

        btnClose.setOnClickListener(new OnClickListener() {
            @Override
            public void onClick(View view) {
                listener.onTabDeleteClicked(tabState);
            }
        });
    }

    public void setListener(Listener listener) {
        this.listener = listener;
    }

    public void bindTabState(WebTabState tabState) {
        this.tabState = tabState;
        tvTitle.setText(TextUtils.isEmpty(tabState.currentTitle) ?
                        Uri.parse(tabState.currentOriginalUrl).getHost() :
                        tabState.currentTitle);
        if (tabState.thumbnail != null) {
            ivThumbnail.setImageBitmap(tabState.thumbnail);
        } else {
            ivThumbnail.setImageResource(android.R.color.transparent);
        }
        llContainer.setBackgroundResource(tabState.selected ?
                R.drawable.selected_tab_button_bg_selector :
                R.drawable.tab_button_bg_selector);
    }

    @Override
    public void onMeasure(int widthMeasureSpec, int heightMeasureSpec) {
        int calculatedHeightSpec = MeasureSpec.makeMeasureSpec(MeasureSpec.getSize(widthMeasureSpec) * 4 / 5, MeasureSpec.getMode(widthMeasureSpec));
        super.onMeasure(widthMeasureSpec, calculatedHeightSpec);
    }

    @Override
    protected void onLayout(boolean changed, int left, int top, int right, int bottom) {
        super.onLayout(changed, left, top, right, bottom);
        listener.onNeededThumbnailSizeCalculated(ivThumbnail.getWidth(), ivThumbnail.getHeight());
    }
}

File: app/src/main/java/com/phlox/tvwebbrowser/activity/main/view/WebTabItemView.kt
package com.phlox.tvwebbrowser.activity.main.view

import android.content.Context
import android.net.Uri
import android.text.TextUtils
import android.view.LayoutInflater
import android.view.View
import android.widget.Button
import android.widget.FrameLayout
import android.widget.ImageView
import android.widget.LinearLayout
import android.widget.TextView

import com.phlox.tvwebbrowser.R
import com.phlox.tvwebbrowser.model.WebTabState
import kotlinx.android.synthetic.main.view_tab_item.view.*

/**
 * Created by PDT on 24.08.2016.
 */
class WebTabItemView(context: Context) : FrameLayout(context) {
    private var tabState: WebTabState? = null
    private var listener: Listener? = null

    interface Listener {
        fun onTabSelected(tab: WebTabState)
        fun onTabDeleteClicked(tab: WebTabState)
        fun onNeededThumbnailSizeCalculated(width: Int, height: Int)
    }

    init {
        LayoutInflater.from(context).inflate(R.layout.view_tab_item, this)

        llContainer.setOnClickListener { listener!!.onTabSelected(tabState!!) }
        btnClose.setOnClickListener { listener!!.onTabDeleteClicked(tabState!!) }
    }

    fun setListener(listener: Listener) {
        this.listener = listener
    }

    fun bindTabState(tabState: WebTabState) {
        this.tabState = tabState
        tvTitle.text = if (TextUtils.isEmpty(tabState.currentTitle))
            Uri.parse(tabState.currentOriginalUrl).host
        else
            tabState.currentTitle
        if (tabState.thumbnail != null) {
            ivThumbnail.setImageBitmap(tabState.thumbnail)
        } else {
            ivThumbnail.setImageResource(android.R.color.transparent)
        }
        llContainer.isActivated = tabState.selected
    }

    public override fun onMeasure(widthMeasureSpec: Int, heightMeasureSpec: Int) {
        val calculatedHeightSpec = MeasureSpec.makeMeasureSpec(MeasureSpec.getSize(widthMeasureSpec) * 4 / 5, View.MeasureSpec.getMode(widthMeasureSpec))
        super.onMeasure(widthMeasureSpec, calculatedHeightSpec)
    }

    override fun onLayout(changed: Boolean, left: Int, top: Int, right: Int, bottom: Int) {
        super.onLayout(changed, left, top, right, bottom)
        listener!!.onNeededThumbnailSizeCalculated(ivThumbnail.width, ivThumbnail.height)
    }
}
